src/lib/support/logging/TextOnlyLogging.h

```diff
@@ -268,16 +268,21 @@ using LogRedirectCallback_t = void (*)(const char * module, uint8_t category, co
 #define ChipLogValueMEI(aValue) static_cast<uint16_t>(aValue >> 16), static_cast<uint16_t>(aValue)
 
 /**
- * Logging helpers for exchanges.  For now just log the exchange id and whether
- * it's an initiator or responder, but eventually we may want to log the peer
- * node id as well (especially for the responder case).  Some callsites only
+ * Logging helpers for exchanges.  Log the exchange id, whether
+ * it's an initiator or responder and the peer node id.  Some callsites only
  * have the exchange id and initiator/responder boolean, not an actual exchange,
  * so we want to have a helper for that case too.
  */
 #define ChipLogFormatExchangeId "%u%c"
 #define ChipLogValueExchangeId(id, isInitiator) id, ((isInitiator) ? 'i' : 'r')
-#define ChipLogFormatExchange ChipLogFormatExchangeId
-#define ChipLogValueExchange(ec) ChipLogValueExchangeId((ec)->GetExchangeId(), (ec)->IsInitiator())
+#define ChipLogFormateExchangeNode "[%u:" ChipLogFormatX64 "]"
+#define ChipLogValueExchangeNode(fabric, id) fabric, ChipLogValueX64(id)
+#define ChipLogFormatExchange ChipLogFormatExchangeId " to: " ChipLogFormateExchangeNode
+#define ChipLogValueExchange(ec) ChipLogValueExchangeId((ec)->GetExchangeId(), (ec)->IsInitiator()),                               \
+    ChipLogValueExchangeNode(                                                                                                      \
+        ((ec)->HasSessionHandle() ? (ec)->GetSessionHandle()->GetSubjectDescriptor().fabricIndex : kUndefinedFabricIndex),         \
+        ((ec)->HasSessionHandle() ? (ec)->GetSessionHandle()->GetSubjectDescriptor().subject : kUndefinedNodeId))   
```

Comment:
So this will have the following behavior:

1. For CASE sessions, what you would expect.
2. For incoming group messages, this will list the group id (as opposed to the source node id from the message).
3. For outgoing group messages, this will list undefined node id.
4. For PASE sessions as initiator, this will list undefined node id.
5. For PASE sessions as responder, this will list the made-up peer node id.
6. For unauthenticated sessions, this will list undefined node id.

The real question is this: is this better than doing `(ec)->GetSessionHandle()->GetPeer()` and logging it with `ChipLogFormatScopedNodeId`?  The latter is probably also smaller code.

Reply:
It looks like the `ScopedNodeId` is used in most other places where the node id is logged, so i have changed my implementation.